app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,334 @@
+            EventsService.markCleared(notification.event);
+          });
+          group.notifications = [];
+          render();
+          $rootScope.$emit('NotificationDrawerWrapper.onMarkAllRead');
+        },
+        notificationGroups: notificationGroups,
+        headingInclude: 'views/directives/notifications/header.html',
+        notificationBodyInclude: 'views/directives/notifications/notification-body.html',
+        customScope: {
+          clear: function(notification, index, group) {
+            EventsService.markCleared(notification.event);
+            group.notifications.splice(index, 1);
+            countUnreadNotificationsForAllGroups();
+          },
+          markRead: function(notification) {
+            notification.unread = false;
+            EventsService.markRead(notification.event);
+            countUnreadNotificationsForAllGroups();
+          },
+          getNotficationStatusIconClass: function(event) {
+            return iconClassByEventSeverity[event.type] || iconClassByEventSeverity.info;
+          },
+          getStatusForCount:  function(countKey) {
+            return iconClassByEventSeverity[countKey] || iconClassByEventSeverity.info;
+          },
+          close: function() {
+            drawer.drawerHidden = true;
+          }
+        }
+      });
+
+      var projectChanged = function(next, current) {
+        return _.get(next, 'params.project') !== _.get(current, 'params.project');
+      };
+
+      var reset = function() {
+        getProject($routeParams.project).then(function() {
+          watchEvents($routeParams.project, eventWatchCallback);
+          //watchNotifications($routeParams.project, notificationWatchCallback);
+          hideIfNoProject($routeParams.project);
+          render();
+        });
+      };
+
+      var initWatches = function() {
+        if($routeParams.project) {
+          reset();
+        }
+        // $routeChangeSuccess seems more reliable than $locationChangeSuccess:
+        // - it fires once on initial load. $locationChangeSuccess does not.
+        // - it waits for more object resolution (not a huge deal in this use case)
+        // - tracks route data instead of urls (args to callback fn, also not
+        //   necessary for the current use case)
+        rootScopeWatches.push($rootScope.$on("$routeChangeSuccess", function (evt, next, current) {
+          if(projectChanged(next, current)) {
+            drawer.customScope.projectName = $routeParams.project;
+            reset();
+          }
+        }));
+
+        // event from the counter to signal the drawer to open/close
+        rootScopeWatches.push($rootScope.$on('NotificationDrawerWrapper.toggle', function() {
+          drawer.drawerHidden = !drawer.drawerHidden;
+        }));
+      };
+
+      drawer.$onInit = function() {
+        if(DISABLE_GLOBAL_EVENT_WATCH || LIMIT_WATCHES) {
+          return;
+        }
+        if (AuthService.isLoggedIn()) {
```

Comment:
@spadgett for reference, updates here.

Reply:
I know I suggested this, but I'd like to check if we can get away with only starting the watch `$onRouteChangeSuccess` and if this is still needed in that case?

Reply:
It will work with just the `$onRouteChangeSuccess`, however, are we sure we don't want to worry about the possibility of future public pages?   I hate to leave that possibility open but unhandled in case we create a future bug.